packages/material-ui/package.json

```diff
@@ -50,6 +50,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.4.4",
+    "@material-ui/styled-engine": "^5.0.0-alpha.1",
```

Comment:
I think this dependency tree is not deterministic. If I install `@material-ui/core` and `@emotion/core` then package managers won't know that `@emotion/core` belongs to `@material-ui/styled-engine`. It'll work in most trees due to hoisting but is technically wrong. `@material-ui/core` would need to hoist the peers (by re-declaring them) of `@material-ui/styled-engine` as well. 

Though we can revisit this once published (actually uploaded to npm) and check how modern package managers (yarn v2 or pnpm) treat this dependency tree.

Reply:
Wasn't aware that this can create a problem, let me add it in the list of things we need to check 👍

Reply:
I can only confirm - peer deps requirements spread across the dependency chain :/ in this case what is a peer dep of `@material-ui/styled-engine` should be a peer dep of the package that depends on `@material-ui/styled-engine`

Reply:
I've added the emotion dependencies as `peerDependencies` for `@material-ui/core`

Reply:
@eps1lon should we also update the `peerDependencies` inside `@material-ui/icons` and `@material-ui/lab` as these packages have the core in the `peerDependencies`?